server/node/src/main/java/org/kaaproject/kaa/server/operations/service/delta/DefaultDeltaService.java

```diff
@@ -208,12 +213,19 @@ public ConfigurationCacheEntry compute(DeltaCacheKey deltaKey) {
                                 userConfigurationHash = EndpointObjectHash.fromString(userConfiguration.getBody());
                             }
 
-                            BaseData mergedConfiguration = getMergedConfiguration(endpointId, userConfiguration, deltaKey,
-                                    latestConfigurationSchema);
+                            BaseData mergedConfiguration = getMergedConfiguration(endpointId, userConfiguration, deltaKey, latestConfigurationSchema);
 
-                            LOG.trace("[{}] Merged configuration {}", endpointId, mergedConfiguration.getRawData());
+                            if(useConfigurationRawSchema) {
+                                kaaData = mergedConfiguration;
+                            } else {
+                                // converting merged base schema to raw schema
+                                String ctlSchema = cacheService.getFlatCtlSchemaById(latestConfigurationSchema.getCtlSchemaId());
```

Comment:
This operation need to be cached. I suggest to modify "getMergedConfiguration" method to return simple structure instead of BaseData. This structure will contain both BaseData and RawData